src/Oro/Bundle/EntityConfigBundle/Migration/RemoveEnumFieldQuery.php

```diff
@@ -97,8 +97,9 @@ protected function deleteEnumData(LoggerInterface $logger, $id, $data)
     protected function updateEntityData(LoggerInterface $logger, $enumClass, $data)
     {
         $data = $data ? $this->connection->convertToPHPValue($data, Type::TARRAY) : [];
+        $relationType = $this->getRelationType();
 
-        $extendKey = sprintf('manyToOne|%s|%s|%s', $this->entityClass, $enumClass, $this->enumField);
+        $extendKey = sprintf($relationType . '|%s|%s|%s', $this->entityClass, $enumClass, $this->enumField);
```

Comment:
looks like $relationType could be an optional parameter of `RemoveEnumFieldQuery::__construct` instead. 
I don't see the reason of creating 2 more classes and brake the backward compatibility here by making RemoveEnumFieldQuery abstract